emb/network/transport/BUILD

```diff
@@ -26,49 +37,95 @@ py_library(
     deps = [requirement("pyzmq")],
 )
 
+pico_btstack_make_gatt_header(
+    name = "nus",
+    src = "nus.gatt",
+)
+
+cc_library(
+    name = "btstack_config",
+    hdrs = ["btstack_config.h"],
+    includes = ["."],  # Shorten the include path to what dependents expect
+    visibility = ["//visibility:public"],
+)
+
 cc_library(
     name = "transporter_cc",
     hdrs = ["transporter.hpp"],
     visibility = ["//visibility:public"],
 )
 
-cc_library(
-    name = "serial_cc",
-    srcs = select(
+alias(
+    name = "transport",
+    actual = select(
         {
-            "@platforms//cpu:armv6-m": ["pico/serial.cc"],
-            "//bzl/platforms:unittest": ["unittest/serial.cc"],
-            "//conditions:default": ["host/serial.cc"],
+            "//bzl/constraints:comms_protocol_ble_enabled": ":transport_ble",
+            "//bzl/constraints:comms_protocol_serial_enabled": ":transport_serial",
+            "//bzl/constraints:comms_protocol_zmq_enabled": ":transport_zmq",
+            "//bzl/constraints:comms_protocol_unittest_enabled": ":transport_unittest",
```

Comment:
I'm a fan of how this turned out; the Pico SDK is a great example. Great cleanup in general for the transport layer. But I'm only halfway there; we can specify one transport layer agnostically with `:transport` & `transport.hpp`, but what if I want a second layer, so log messages can pass through `serial` instead of `ble`? Stuck on this one still